### ISSUE TITLE
Code-folding only works for some languages

### ISSUE THREAD

**Issue by msbentley** [BoostIO/Boostnote#1823]
# Current behavior

Code folding does not seem to work in the snippet editor for all languages. If I compare to the examples given at: https://codemirror.net/demo/folding.html then I get:

- python - no folding indicators shown
- HTML - 2 indicators shown (lines 7 and 30)
- JavaScript - works fine (although I'm pretty sure I tried this yesterday and it *didn't* work...)

Folding also seems to work for JSON, perhaps others but I haven't tested further...

# Expected behavior

Code folding is available for all languages supported by CodeMirror.

# Steps to reproduce

1. Create a new snippet
2. Set the language accordingly
3. Look for folding markers in the margins

# Environment

- Version: 0.11.4
- OS Version and name: ubuntu 16.04

![boostnote_021](https://user-images.githubusercontent.com/1671065/38875787-7994b0be-425b-11e8-96c3-d300ee22b871.png)

**Comment by gediminasml** [comment]
What's worth investigating - whether `mode` parameter (f.e. `mode: "python"`) is set correctly while initializing `CodeMirror` library.

**Comment by Flexo013** [comment]
@msbentley @gediminasml Is this still an issue in the latest version?

**Comment by msbentley** [comment]
Yes, it is still an issue on 0.12.1-1 - testing quickly with the example from the CodeMirror [code-folding page](https://codemirror.net/demo/folding.html) js works fine, but python and html do not, for example.
